src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -402,14 +402,8 @@ private void BuildProjectWithMsbuild()
             string properties = string.Empty;
             foreach (var property in ProjectProperties)
             {
-                if (property.Value.Contains(" "))
-                {
-                    properties += $" /p:{property.Key}=\"{property.Value}\"";
-                }
-                else
-                {
-                    properties += $" /p:{property.Key}={property.Value}";
-                }
+                string escapedValue = MsBuildUtility.Escape(property.Value);
+                properties += $" /p:{property.Key}={escapedValue}";
```

Comment:
Does the key ever need to be escaped?
nit: properties will have an extra space at the beginning, meaning the parameters below will have a double space.